Assets/Scripts/GameplayObjects/Roads/RoadBaseFactory.cs

```diff
@@ -0,0 +1,8 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class RoadBaseFactory
```

Comment:
this class seems redundant. you inherit it once, and you don't use the abstraction. you should delete it